signer-cloud-server/src/main/java/com/wultra/signercloud/server/document/DocumentService.java

```diff
@@ -188,17 +203,164 @@ private static byte[] getFileBytes(final MultipartFile file) {
+     * into {@link DocumentContent} (It overrides the original unsigned PDF content with the signed one.).
+     *
+     * @param documentId identifier of the document to be signed
+     * @param requestBody request body containing the signature
+     * @return response as a {@link Try}
+     */
+    Try<SignDocumentResponse> signDocument(final String documentId, final SignDocumentRequest requestBody) {
+        try {
+            final var response = processSignDocument(documentId, requestBody);
+            return Try.success(response);
+        } catch (final CertificateException | DocumentNotFoundException | SignerNotFoundException | SignDocumentException e) {
+            return Try.error(e);
+        }
+    }
+
+    private SignDocumentResponse processSignDocument(final String documentId, final SignDocumentRequest requestBody) throws CertificateException {
+
+        final var document = documentRepository.findByDocumentId(documentId)
+                .orElseThrow(() -> new DocumentNotFoundException("Document not found for document ID: " + documentId));
+
+        final var documentContent = documentContentRepository.findById(document.getDocumentContentId())
+                .orElseThrow(() -> new DocumentNotFoundException("Document content not found for document ID: " + documentId));
+
+        final var signer = signerRepository.findById(document.getSignerId())
+                .orElseThrow(() -> new SignerNotFoundException("Signer not found for document ID: " + documentId));
+
+        verifyDocumentCanBeSigned(signer, document);
+
+        final var signature = requestBody.signature();
+        final var signedDocumentBytes = verifySignatureAndSignDocument(signer.getCertificate(),
+                document.getHash(),
+                signature,
+                documentContent.getContent(),
+                configurationProperties.getSignatureHashAlgorithm());
+
+        final var updatedDocumentContent = documentContent.toBuilder()
+                .content(signedDocumentBytes)
+                .build();
+
+        documentContentRepository.save(updatedDocumentContent);
+
+        final var updatedDocument = document.toBuilder()
+                .timestampLastUpdated(Instant.now())
+                .fileSize(signedDocumentBytes.length)
+                .status(DocumentStatus.SIGNED)
+                .signature(signature)
+                .build();
+
+        documentRepository.save(updatedDocument);
+
+        final var downloadUrl = buildDocumentDownloadUri(documentId);
+        return new SignDocumentResponse(documentId, downloadUrl);
+    }
+
+    private void verifyDocumentCanBeSigned(final Signer signer, final Document document) {
+        if (SignerStatus.ACTIVE != signer.getStatus()) {
+            throw new SignDocumentException("Signer is not active. Signer: " + signer.getExternalSignerId());
+        }
+
+        if (DocumentStatus.WAITING != document.getStatus()) {
```

Comment:
Personally, I would reorder it, value first, constant second.
```suggestion
        if (document.getStatus() != DocumentStatus.WAITING) {
```

Reply:
Updated.